pallets/collator-selection/src/lib.rs

```diff
@@ -377,11 +379,11 @@ pub mod pallet {
 		pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
 	{
 		fn note_author(author: T::AccountId) {
-			let treasury = Self::account_id();
-			let reward = T::Currency::free_balance(&treasury).div(2u32.into());
-
-			// `reward` is half of treasury account, this should never fail.
-			let _success = T::Currency::transfer(&treasury, &author, reward, KeepAlive);
+			let pot = Self::account_id();
+			// assumes an ED will be sent to pot.
+			let reward = T::Currency::free_balance(&pot).checked_sub(&T::Currency::minimum_balance()).unwrap_or_else(Zero::zero).div(2u32.into());
```

Comment:
Why is this using checked_sub + unwrap?

You can just saturating sub?

Reply:
You could, but IMO checked is more expressive of the operation. To configure the staking pot, someone has to send the ED to it.

There's no case where `0 < balance < ED`, so using checked and getting `None` back is saying that this hasn't been initialized. I see saturating (and getting zero) as saying that there's just not enough, but that doesn't express the binary-ness of the state of the pot.

Reply:
Sorry but this doesnt make sense to me.

Either the subtraction will work and you will have a value, or the subtraction will fail, and you want zero.

This is precisely what saturating_sub does. If you actually did something with the result, that would be one thing, but since you just pick zero when the subtraction fails, saturating is clearly the more ergonomic tool for this.